### DISCUSSION TITLE
Dashboard roadmap update and question: What does 'external contributors' mean to you?

### DISCUSSION THREAD

**Discussion by ahpook** [community/OSPO#35] [Dashboard]
Hi all, I wanted to drop an update here and pose a question to you. 

First off, we're hard at work adding the number one most requested metric to the dashboard: pull requests! Specifically, we're going to be able to show the number of open vs closed pull requests over time as well as how many ended up merged vs being closed without merge. Adding this involves both crunching the numbers on the back end and displaying some new visualizations but we plan to get it out before the quarter's up (end of march).

In parallel with that work, we're looking into what it'll take to open up the dashboard to more orgs  and ease the onboarding experience. An improvement to the data layer shipped this week (Jan 24) that will greatly improve the performance for large orgs - especially ones who've reported their graphs not loading because they timed out.

Looking forward, one impediment the engineers have identified is the complexity of the query behind the "External Contributors" metric. For background, right now this presents repo activity as a percentage, by users who are members of the GitHub Organization that owns the repository vs those who have write access but are not members. The idea was to align with the [CHAOSS Project 'contributors' metric](https://chaoss.community/kb/metric-contributors/), where an increasing number of non-member contributors can indicate a healthy, growing community of new users.

But we've already heard some feedback that our implementation of this metric doesn't align with how some projects think about 'core' vs 'external' contributors, because organization membership isn't the domain boundary; instead, it's commit right to the repository. So the notion that 'external contributors' is not especially useful, plus imposes scaling limitations as implemented, makes me wonder whether we'd be losing anything of value by dropping it. 

What do you think? Is it important to you, and if so, what do you use it for? Is there another metric we should be using instead? We'd love your input.

**Comment by jeffwilcox** [comment]
Such an interesting place...

Selfishly, I wish that for a given project, somehow _I could provide_ an API or private way for you to understand what I think about a given project, or, at least have some kind of config-as-code file to help indicate who "core" maintainers are.

If I think about the Microsoft example, not all of our employees publicize their membership in the Microsoft official organization. We ask that people comply with our company social media policies, and indicate clearly they work at Microsoft, but that can take many forms:

- Mention the company in their bio
- Publicize their Microsoft org membership
- Company name or at-mention on profile
- README.md in profile
- Mentioned in pull requests and/or Git commit author fields

So I have a list of 50,000+ people at my company who have chosen to tell us who they are on GitHub; however, it's not exactly a list we could make public, and it's changing all the time.

So if an employee contributed to a project 5 years ago, and since has left for a competitor, I think that contribution at the time was probably a company/project contributor, and now, no longer such a contribution...

**Comment by emmairwin** [comment]
To me this is about enabling attestation of contributions.  Of which external contributors might be
- partners and potential partners
- Developers and others on behalf of customers
- Developers and others on behalf of their employer
- Students
- Fans and champions

I think the CHAOSS metric more closely associated with this would be: https://chaoss.community/kb/metric-contribution-attribution/

The audiences who care about each would be
- OSPO  : As a user I would like to help our product teams evaluate ecosystem engagement for business decisions.  As a user I would like to help our employees highlight strategic contributions for their review cycles.
- GitHub orgs/Companies:  As a user I would like to highlight the impact of our employees on the ecosystem, or specific project/technology areas.
- Products/Projects: as a user I would like to know what kinds of projects and in which ecosystems our projects is being deployed, to help us make better investment decisions.
- Contributor: As a contributor I want to reflect my marketable accomplishments to build a reputation (beyond the graph)
